### PR TITLE
Implement nanosleep(), fixes in sleep() and usleep()

### PR DESCRIPTION
**Pull request by agkaminski**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (ia32-generic).

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (is dependent on https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/337).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: time/time.c [hunk 1]
```diff
@@ -422,7 +422,21 @@ char *strptime(const char *restrict buf, const char *restrict format, struct tm
...
 int nanosleep(const struct timespec *req, struct timespec *rem)
 {
-	return 0;
+	time_t sec = req->tv_sec;
+	long nsec = req->tv_nsec;
+	int ret;
+
+	ret = nsleep(&sec, &nsec);
+
+	if (ret == -EINTR) {
```
Comment (nalajcie) [inline time/time.c:436]: check if `rem` is not null (allowed by specs)

File: unistd/sys.c [hunk 2]
```diff
@@ -246,9 +246,34 @@ int execle(const char *path, const char *arg, ...)
 }
 
 
+extern int nsleep(time_t *sec, long *nsec);
+
+
+extern int usleep(useconds_t usecs)
```
Comment (nalajcie) [inline unistd/sys.c:252]: no need for extern

**Review by nalajcie — CHANGES_REQUESTED** [review]
check for NULL please ;)

File: time/time.c (hunk 1 above)
Comment (agkaminski) [reply time/time.c]: Done

File: unistd/sys.c (hunk 2 above)
Comment (agkaminski) [reply unistd/sys.c]: Whops, copy-paste error, thanks

**Review by nalajcie — APPROVED** [review]

**Pull request merged**